#!/usr/bin/env python
PACKAGE = "hector_color_detection_nodelet"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_H_value", int_t, 0, "Minimum hue value", 100, 0, 179)
gen.add("max_H_value", int_t, 0, "Maximum hue value", 140, 0, 179)
gen.add("min_S_value", int_t, 0, "Minimum saturation value", 100, 0, 255)
gen.add("max_S_value", int_t, 0, "Maximum saturation value", 255, 0, 255)
gen.add("min_V_value", int_t, 0, "Minimum value value", 50, 0, 255)
gen.add("max_V_value", int_t, 0, "Maximum value value", 200, 0, 255)
gen.add("bluePart", double_t, 0, "blue part of the image", 4, 0, 20)
gen.add("minRadius", double_t, 0, "Minimum radius", 0.15, 0.0, 1.0)
gen.add("maxRadius", double_t, 0, "Maximum radius", 0.4, 0.1, 1.0)

exit(gen.generate(PACKAGE, "hector_color_detection_nodelet", "ColorDetection"))
